homeassistant/components/climate/zha.py

```diff
@@ -0,0 +1,251 @@
+                                       foundation.Discrete)
+        foundation.DATA_TYPE_IDX[t.enum8] = 0x30
+        self._endpoint.thermostat.attributes[0x001C] = \
+            ('system_mode', t.enum8)
+        # END
+
+        # pylint: disable=W0212
+        self._endpoint.thermostat._attridx['setpoint_mode_manuf_specific'] = \
+            0x401C
+        # pylint: enable=W0212
+        self._endpoint.thermostat.attributes[0x401C] = \
+            ('setpoint_mode_manuf_specific', t.enum8)
+
+    @asyncio.coroutine
+    def _wait_for_clean_communication(self):
+        """Ensure that we not request the device more than time by second."""
+        if time.time() - self.last_request < CLEAN_TIME_ZONE:
+            yield from asyncio.sleep(CLEAN_TIME_ZONE)
+        self.last_request = time.time()
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+        _LOGGER.debug("%s async_update", self.entity_id)
+        yield from self._wait_for_clean_communication()
+        result = yield from safe_read(self._endpoint.thermostat,
+                                      ['local_temp',
+                                       'pi_heating_demand',
+                                       'occupied_heating_setpoint',
+                                       'setpoint_mode_manuf_specific'])
+        self.last_request = time.time()
+        if not result:
+            _LOGGER.warning("No result received")
+            return
+
+        try:
+            self._heating_power = int(result['pi_heating_demand'])
+        except (TypeError, KeyError):
+            _LOGGER.warning("Can not get heating power value")
+            self._heating_power = None
+
+        raw_mode = result.get('setpoint_mode_manuf_specific')
+        if raw_mode not in OPERATIONS:
+            _LOGGER.warning("Mode %s not found in known modes %s",
+                            raw_mode, OPERATIONS)
+        self._current_operation = OPERATIONS[raw_mode]
+
+        raw_temp = result.get('occupied_heating_setpoint')
+        if raw_temp:
+            self._target_temperature = raw_temp / 100.0
+        raw_temp = result.get('local_temp')
+        if raw_temp:
+            self._current_temperature = raw_temp / 100.0
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """Return a list of available operation modes."""
+        return OPERATION_LIST
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self._target_temperature is None or self._target_temperature < 0:
```

Comment:
How can `self._target_temperature < 0` happen? Is that used as error indicator somewhere? I can't find it.